docs/core/extensions/httpclient-factory-keyed-di.md

```diff
@@ -0,0 +1,302 @@
+    - Plain Transient service `Service`.
+2. `HttpClient` dependency in `Service` is explicitly bound to a Keyed Service with a key `nameof(Service)`.
+
+The name doesn't have to be `nameof(Service)`, but the example aimed to minimize the behavioral changes. Internally, typed clients use Named clients, and by default, such "hidden" Named clients go by the linked Typed client's type name. In this case, the "hidden" name was `nameof(Service)`, so the example preserved it.
+
+Technically, the example "unwraps" the Typed client, so that the previously "hidden" Named client becomes "exposed," and the dependency is satisfied via the Keyed DI infra instead of the Typed client infra.
+
+## How to: Opt in to Keyed DI by default
+
+You don't have to call <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddAsKeyed%2A> for every single client&mdash;you can easily opt in "globally" (for any client name) via <xref:Microsoft.Extensions.DependencyInjection.HttpClientFactoryServiceCollectionExtensions.ConfigureHttpClientDefaults%2A>. From Keyed Services perspective, it results in the <xref:Microsoft.Extensions.DependencyInjection.KeyedService.AnyKey?displayProperty=nameWithType> registration.
+
+```csharp
+services.ConfigureHttpClientDefaults(b => b.AddAsKeyed());
+
+services.AddHttpClient("first", /* ... */);
+services.AddHttpClient("second", /* ... */);
+services.AddHttpClient("third", /* ... */);
+
+public class MyController(
+    [FromKeyedServices("first")] HttpClient first,
+    [FromKeyedServices("second")] HttpClient second,
+    [FromKeyedServices("third")] HttpClient third)
+//{ ...
+```
+
+### Beware Of "Unknown" clients
+
+> [!NOTE]
+> `KeyedService.AnyKey` registrations define a mapping from _any_ key value to some service instance. However, as a result, the Container validation doesn't apply, and an _erroneous_ key value _silently_ leads to a _wrong instance_ being injected.
+
+> [!IMPORTANT]
+> For Keyed `HttpClient`s, a mistake in the client name can result in erroneously injecting an "unknown" client&mdash;meaning, a client whose name was never registered.
+
+The same is true for the plain Named clients: `IHttpClientFactory` doesn't require the client name to be explicitly registered (aligning with the way the [Options pattern](options.md) works). The factory gives you an unconfigured&mdash;or, more precisely, default-configured&mdash;`HttpClient` for any unknown name.
+
+> [!NOTE]
+> Therefore, it's important to keep in mind: the "Keyed by default" approach covers not only all _registered_ `HttpClient`s, but all the clients that `IHttpClientFactory` is _able to create_.
+
+```csharp
+services.ConfigureHttpClientDefaults(b => b.AddAsKeyed());
+services.AddHttpClient("known", /* ... */);
+
+provider.GetRequiredKeyedService<HttpClient>("known");   // OK
+provider.GetRequiredKeyedService<HttpClient>("unknown"); // OK (unconfigured instance)
```

Comment:
This is very similar to the example from the beginning of the document and I'd explicitly call out the difference and why it doesn't throw here.